### PR TITLE
Swagger Documentation for CollateralsApiResource.java and InterestRateChartsApiResource.java

### PR DESCRIPTION
**Pull request by luckyman20**

### CODE REVIEW

**Comment by Sanyam96** [comment]
Naming Convention is not correct!

**Comment by luckyman20** [comment]
I will correct that.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/collateral/api/CollateralsApiResource.java [hunk 1]
```diff
@@ -116,8 +110,10 @@ public String retrieveCollateralDetails(@Context final UriInfo uriInfo, @PathPar
     @Path("{collateralId}")
     @Consumes({ MediaType.APPLICATION_JSON })
     @Produces({ MediaType.APPLICATION_JSON })
+    @ApiOperation(value = "Retrieve a Collateral", httpMethod = "GET", notes = "Example Requests:\n" + "\n" + "/loans/1/collaterals/1\n" + "\n" + "\n" + "/loans/1/collaterals/1?fields=value,description")
+    @ApiResponses({@ApiResponse(code = 200, message = "OK", response = CollateralsApiResourceSwagger.GetLoansLoanIdCollateralsResponse.class)})
     public String retrieveCollateralDetails(@Context final UriInfo uriInfo, @PathParam("loanId") final Long loanId,
```
Comment (Sanyam96) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/collateral/api/CollateralsApiResource.java]: @ApiParam annotation is missing

**Review by Sanyam96 — APPROVED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/collateral/api/CollateralsApiResource.java [hunk 2]
```diff
@@ -162,7 +164,9 @@ public String updateCollateral(@PathParam("loanId") final Long loanId, @PathPara
     @Path("{collateralId}")
     @Consumes({ MediaType.APPLICATION_JSON })
     @Produces({ MediaType.APPLICATION_JSON })
-    public String deleteCollateral(@PathParam("loanId") final Long loanId, @PathParam("collateralId") final Long collateralId) {
+    @ApiOperation(value = "Remove a Collateral", httpMethod = "DELETE", notes = "Note: A collateral can only be removed from Loans that are not yet approved.")
+    @ApiResponses({@ApiResponse(code = 200, message = "OK", response = CollateralsApiResourceSwagger.PostLoansLoanIdCollateralsCollateralIdResponse.class)})
+    public String deleteCollateral(@PathParam("loanId") @ApiParam(value = "loanId") final Long loanId, @PathParam("collateralId") @ApiParam(value = "collateralId")final Long collateralId) {
```
Comment (Sanyam96) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/collateral/api/CollateralsApiResource.java]: response name of inner class is not correct

**Comment by luckyman20** [comment]
@Sanyam96 I have updated the PR

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/collateral/api/CollateralsApiResource.java [hunk 3]
```diff
@@ -162,7 +164,9 @@ public String updateCollateral(@PathParam("loanId") final Long loanId, @PathPara
     @Path("{collateralId}")
     @Consumes({ MediaType.APPLICATION_JSON })
     @Produces({ MediaType.APPLICATION_JSON })
-    public String deleteCollateral(@PathParam("loanId") final Long loanId, @PathParam("collateralId") final Long collateralId) {
+    @ApiOperation(value = "Remove a Collateral", httpMethod = "DELETE", notes = "Note: A collateral can only be removed from Loans that are not yet approved.")
+    @ApiResponses({@ApiResponse(code = 200, message = "OK", response = CollateralsApiResourceSwagger.DeleteLoansLoanIdCollateralIdCollatarealsResponse.class)})
```
Comment (Sanyam96) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/collateral/api/CollateralsApiResource.java]: naming of inner class is  not correct

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/collateral/api/CollateralsApiResource.java [hunk 4]
```diff
@@ -135,7 +131,10 @@ public String retrieveCollateralDetails(@Context final UriInfo uriInfo, @PathPar
     @POST
     @Consumes({ MediaType.APPLICATION_JSON })
     @Produces({ MediaType.APPLICATION_JSON })
-    public String createCollateral(@PathParam("loanId") final Long loanId, final String apiRequestBodyAsJson) {
+    @ApiOperation(value = "Create a Collateral", httpMethod = "POST", notes = "Note: Currently, Collaterals may be added only before a Loan is approved")
+    @ApiImplicitParams({@ApiImplicitParam(value = "body", required = true, paramType = "body", dataType = "body", format = "body", dataTypeClass = CollateralsApiResourceSwagger.PostLoansLoanIdCollateralsRequest.class)})
+    @ApiResponses({@ApiResponse(code = 200, message = "OK", response = CollateralsApiResourceSwagger.PostLoansLoanIdCollateralsCollateralIdResponse.class)})
```
Comment (Sanyam96) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/collateral/api/CollateralsApiResource.java]: check the naming of the inner class

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/collateral/api/CollateralsApiResourceSwagger.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+import java.util.Set;
+
+/**
+ * Created by Chirag Gupta on 12/01/17.
+ */
+public class CollateralsApiResourceSwagger {
```
Comment (Sanyam96) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/collateral/api/CollateralsApiResourceSwagger.java]: class should not be public
make it final

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/collateral/api/CollateralsApiResource.java (hunk 4 above)
Comment (luckyman20) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/collateral/api/CollateralsApiResource.java]: what should be the correct name then?

Comment (Sanyam96) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/collateral/api/CollateralsApiResource.java]: PostLoansLoanIdCollateralsResponse

**Review by Sanyam96 — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/collateral/api/CollateralsApiResource.java [hunk 6]
```diff
@@ -162,7 +164,9 @@ public String updateCollateral(@PathParam("loanId") final Long loanId, @PathPara
     @Path("{collateralId}")
     @Consumes({ MediaType.APPLICATION_JSON })
     @Produces({ MediaType.APPLICATION_JSON })
-    public String deleteCollateral(@PathParam("loanId") final Long loanId, @PathParam("collateralId") final Long collateralId) {
+    @ApiOperation(value = "Remove a Collateral", httpMethod = "DELETE", notes = "Note: A collateral can only be removed from Loans that are not yet approved.")
+    @ApiResponses({@ApiResponse(code = 200, message = "OK", response = CollateralsApiResourceSwagger.DeleteLoansLoanIdCollatarealsCollateralIdResponse.class)})
```
Comment (Sanyam96) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/collateral/api/CollateralsApiResource.java]: Check the spelling of inner class

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/interestratechart/api/InterestRateChartsApiResource.java [hunk 7]
```diff
@@ -141,7 +134,9 @@ public String retrieveOne(@PathParam("chartId") final Long chartId, @Context fin
     @POST
     @Consumes({ MediaType.APPLICATION_JSON })
     @Produces({ MediaType.APPLICATION_JSON })
-    public String create(final String apiRequestBodyAsJson) {
+    @ApiOperation(value = "Create a Chart", httpMethod = "POST", notes = "Creats a new chart which can be attached to a term deposit products (FD or RD).")
+    @ApiImplicitParams({@ApiImplicitParam(value = "body", required = true, paramType = "body", dataType = "body", format = "body", dataTypeClass = InterestRateChartsApiResourceSwagger.PostInterestRateChartsRequest.class)})
+    public String create(@ApiParam(hidden = true) final String apiRequestBodyAsJson) {
```
Comment (Sanyam96) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/interestratechart/api/InterestRateChartsApiResource.java]: The response for this method is missing!

**Comment by Sanyam96** [comment]
Response of GET https://DomainName/api/v1/interestratecharts/{chartId}
is not correct.

**Review by Sanyam96 — APPROVED** [review]

**Pull request merged**